core/src/main/scala/org/broadinstitute/dsde/rawls/workspace/WorkspaceService.scala

```diff
@@ -1565,6 +1634,39 @@ class WorkspaceService(protected val ctx: RawlsRequestContext,
         ErrorReport(StatusCodes.BadRequest, "may not grant readers compute access")
       )
     }
+    if (
```

Comment:
Alternatively, we can not throw if someone tries to grant a reader or writer share/compute access on an McWorkspace and just silently not grant them that permission. I figured failing loudly was better than not granting the permissions that had been requested

Reply:
Agreed, that is a programming error we should not mask.